### PR TITLE
Version 1.0 Recipe App

### PR DESCRIPTION
**Pull request by indigodavid**
# Version 1.0 Recipe App 🎉 

This version includes the following features:

## Public Recipes

- Every public recipe is accesible to any user, even guests.
- Each user can choose if their recipe is public or no

## User

- The app allows users to be created and edited
- Users can manage their own resources => `Foods`, `Recipes`

## Foods

- Users can create their Foods which represent their own inventory
- Users can manage their own Foods
- These Foods can be used in Recipes

## Recipes

- Users can create their own Recipes
- Once their recipes are created they can add ingredients to their Recipes using their food inventories
- If the ingredient is not present they can create a new one

### Extra Feature: Generate Shopping List
- Users can generate the shopping List for the recipe they are checking

## RecipeFoods

- This is a class that allows to set the quantity of a given Food in a Recipe
- Allows Recipes and Foods to have many-to-many relationship

## Other features

- Used Bootstrap for Styling
- Used Bullet to identify and fix N+1 queries
- Used Devise for Atuthentication
- Used CanCanCan for Authorization
- Used RSpec for testing

### CODE REVIEW

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<p>Welcome <%= @email %>!</p>
+<p>You can confirm your account email through the link below:</p>
+<%# <p><%= link_to 'Confirm my account', confirmation_url(@resource, confirmation_token: @token) %></p> %>
```
[the remark this answers is not in the document]
Comment (indigodavid) [reply app/views/devise/mailer/confirmation_instructions.html.erb]: @codepantha We did the requested change as mentioned. Thank you for letting us know.

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, published: true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
+  end
```
[the remark this answers is not in the document]
Comment (indigodavid) [reply app/models/ability.rb:37]: @codepantha thank you for the observation. We changed the authorizations. It's working now. 
image

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (codepantha) [reply app/views/devise/mailer/confirmation_instructions.html.erb]: You're welcome @indigodavid

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey TEAM 😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**